xoa/resource_xenorchestra_vm.go

```diff
@@ -833,8 +833,14 @@ func resourceVmUpdate(d *schema.ResourceData, m interface{}) error {
 func resourceVmDelete(d *schema.ResourceData, m interface{}) error {
 	c := m.(client.XOClient)
 
-	err := c.DeleteVm(d.Id())
-
+	vmReq := client.Vm{
+		Id: d.Id(),
+	}
+	err := c.HaltVm(vmReq)
```

Comment:
Have you tested what will happen if a VM is already shutdown? I'm thinking about how this would behave if the `resourceVmDelete` were to partially fail or timeout in the halting step.

Reply:
I have updated it so we only shutdown running vms.